daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -547,7 +569,9 @@ class VariableMap private(vTable: Map[GlobalQName, ArrayBuffer[VariableInstance]
         }
 
         case _ => {
-          variable.setValue(VariableUtils.convert(newValue.getAnyRef.toString, variable.rd, referringContext))
+          val value = VariableUtils.convert(newValue.getAnyRef.toString, variable.rd, referringContext)
+          variable.externalValue = value
```

Comment:
I think this ``variable.externalValue = value`` goes away if the logic I mentioned previously makes sense to switch to. Again, one less place where we need to remember to change state.